### ISSUE TITLE
Either add rust support to build-tools, build-tools-proxy, or make a build-tools-rust

### ISSUE THREAD

**Issue by howardjohn** [istio/ztunnel#4]

**Comment by zirain** [comment]
what's your os ? I'm failed to build with following output
```
cargo build
   Compiling serde v1.0.145
   Compiling boring-sys v2.1.0
   Compiling h2 v0.3.14
   Compiling tower v0.4.13
   Compiling ztunnel v0.0.0 (/root/go/src/istio.io/ztunnel)
error: failed to run custom build command for `boring-sys v2.1.0`
```

**Comment by howardjohn** [comment]
the boring ssl issues depend on libclang-dev currently, looking into how to
drop that dependency

**Comment by zirain** [comment]
maybe my ubuntu(Ubuntu 18.04.6 LTS for envoy dev) is too old, still not working

```
Setting up libclang-6.0-dev (1:6.0-1ubuntu2) ...
Setting up libclang-dev (1:6.0-41~exp5~ubuntu1) ...
Processing triggers for libc-bin (2.27-3ubuntu1.6) ...
root@ecs-zirain:~/go/src/istio.io/ztunnel# make build
cargo build
   Compiling serde_json v1.0.86
   Compiling hyper v0.14.20
   Compiling boring-sys v2.1.0
   Compiling tower-http v0.3.4
   Compiling ztunnel v0.0.0 (/root/go/src/istio.io/ztunnel)
error: failed to run custom build command for `boring-sys v2.1.0`
```

**Comment by zirain** [comment]
fixed by upgrade CMake to `3.20.0`
